blog/2022-11-01-cfd-protocol-explained.md

```diff
@@ -0,0 +1,229 @@
+During `contract_setup`, they created and signed `commitment_transaction_0` and a whole set of CETs along with it.
+During `rollover`, they produce a new `commitment_transaction_1` and a new collection of CETs based on a new `announcement` from `olivia`.
+Since they have the means to publish fully signed `commitment_transaction`s, they can eventually publish `commitment_transaction_1` and settle the CFD based on a newer `attestation` from `olivia`.
+
+##### Punish transaction
+
+But, what happens if either party decides to publish an old `commitment_transaction_0` and settle the CFD based on an old BTC price attested by the oracle?
+That would defeat the purpose of rolling over the CFD.
+This is where the revocation mechanism comes in.
+
+A `commitment_transaction`'s `dlc_output` can actually be spent in one more way.
+In full, the `commitment_transaction` can be spent with:
+
+    1. A series of CETs unlocked by one of `olivia`'s `attestation`s.
+    2. A `refund_transaction` that can only be published at least some time after `olivia` has published the relevant `attestation`.
+    3. One `punishment_transaction` per party, which can be constructed unilaterally if one's counterparty publishes a revoked `commitment_transaction`.
+    This transaction allows the aggrieved party.
+
+During `contract_setup` and every subsequent `rollover`, parties work together to add another spending condition to the `dlc_output`.
+They each generate a revocation keypair `(r, R)` and a publication keypair `(p, P)` and share the public keys with each other.
+The `dlc_output` is no longer a simple 2-of-2 multisig on their public keys `A` and `B`.
+It also includes:
+
+    1. A way for the maker to spend it with knowledge of `a`, `r_maker` and `p_maker`.
+    1. A way for the taker to spend it with knowledge of `b`, `r_taker` and `p_taker`.
+
+Instead of exchanging regular signatures on the `commitment_transactions` with each other, each constructs an adaptor signature which when encrypted and published would leak the other party's publishing secret key.
+For example, the taker encrypts under `P_maker` their signature for their public key `B` on the `commitment_transaction`.
+
+Additionally, every time after a contract `rollover` is completed, both parties must exchange their revocation _secret_ keys.
+With all this in place, maker and taker know that they are protected against a counterparty who attempts to publish a revoked (not produced by the most recent `rollover` or `contract_setup`) `commitment_transaction`.
+As soon as they spot it on the blockchain, they can learn the other party's publishing secret key `p`, take the corresponding counterparty revocation secret key `r` that they learnt after rolling over and construct and sign a transaction which gives them all the coins in the `dlc_output`.
```

Comment:
If this is true then I think there is a mixup on line 176 and 177 where we have `r_maker` and `p_maker` on the same spending path? Shouldn't one of those belong to the other?

Reply:
I think the error is in 176 and 177. To punish you need to know: your own signing key: the counterparty's revocation key, learnt after rolling over: and the counterparty's publishing key, leaked when they try to cheat.

So I think this section is right. I'll fix the other one!